### PR TITLE
Split tools.py into separate files under tools/

### PR DESCRIPTION
**Pull request by FabioLuporini**
Besides improving the organisation of the various support functions and classes, this PR removes an annoying dependence between tools and ``configuration``.

### CODE REVIEW

File: devito/tools/utils.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        # the format c_X_p, where X is the C typename, for instance `int` or `float`.
+        return ctype.__name__[2:-2]
+    else:
+        raise TypeError('Unrecognised %s during converstion to C type' % str(ctype))
+
+
+def ctypes_pointer(name):
+    """Create a ctypes type representing a C pointer to a custom data type ``name``."""
+    return type("c_%s_p" % name, (ctypes.c_void_p,), {})
+
```
Comment (mloubout) [inline devito/tools/utils.py]: shouldn't these three be in `types` or something?

Comment (FabioLuporini) [reply devito/tools/utils.py]: I **think** this is a lower level thing

Comment (mloubout) [reply devito/tools/utils.py]: Yes it is, just seem a bit out of place there. I'll leave it to you to decide. GTG otherwise

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**